File: Data Structures & Algorithms/meeting-schedule/submission-2.cs
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

public class Solution {
    public bool CanAttendMeetings(List<Interval> intervals) {
        intervals.Sort((i1, i2) => i1.Start.CompareTo(i2.Start));

        for (int i = 1; i < intervals.Count; i++) {
            Interval i1 = intervals[i - 1];
            Interval i2 = intervals[i];

            if (i1.End > i2.Start) {
                return false;
            }
        }

        return true;
    }
}

File: Data Structures & Algorithms/non-overlapping-intervals/submission-2.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort()

        cnt = 0
        start = 0
        end = -1e99
        for itv in intervals:
            if itv[0] < end:
                cnt += 1
                end = itv[1] if itv[1] < end else end
            else:
                end = itv[1]

        return cnt